### PR TITLE
Fix zeroization issues.

### PR DESCRIPTION
**Pull request by dkasak**

### CODE REVIEW

File: src/olm/session/mod.rs [hunk 1]
```diff
@@ -437,12 +439,15 @@ impl Session {
                 }
 
                 if let Some(chain) = pickle.sender_chains.get(0) {
+                    // XXX: Passing in secret array as value.
```
Comment (poljar) [inline src/olm/session/mod.rs:442]: Can we make these a bit more descriptive. If someone else comes along, they won't have a clue what those are about.

File: src/olm/account/one_time_keys.rs [hunk 2]
```diff
@@ -32,6 +32,14 @@ pub(super) struct OneTimeKeys {
     pub key_ids_by_key: HashMap<Curve25519PublicKey, KeyId>,
 }
 
+impl Drop for OneTimeKeys {
```
Comment (poljar) [inline src/olm/account/one_time_keys.rs:35]: Is this needed? All the values in the map will get dropped and zeroize is implemented for the type in the map.

File: src/megolm/ratchet.rs [hunk 3]
```diff
@@ -208,22 +208,21 @@ pub(super) struct RatchetPickle {
...
-        Self { key: ratchet.inner.into(), counter: ratchet.counter }
+        Self { key: (*ratchet.inner).into(), counter: ratchet.counter }
     }
 }
 
 impl TryFrom<RatchetPickle> for Ratchet {
     type Error = MegolmRatchetUnpicklingError;
 
     fn try_from(pickle: RatchetPickle) -> Result<Self, Self::Error> {
-        Ok(Ratchet {
```
Comment (poljar) [inline src/megolm/ratchet.rs:219]: Need to remember that we need to take another look at this after the simplified pickling branch is merged.

File: src/olm/session/mod.rs [hunk 4]
```diff
@@ -327,8 +327,8 @@ impl Session {
         #[derive(Debug, Zeroize)]
         #[zeroize(drop)]
         struct ReceivingChain {
```
Comment (poljar) [inline src/olm/session/mod.rs:329]: Didn't you intend to rename those so it's clearer they are libolm specific?

**Review by poljar — CHANGES_REQUESTED** [review]
Looks good, left a couple of nits.

File: src/olm/session/mod.rs (hunk 4 above)
Comment (dkasak) [reply src/olm/session/mod.rs:329]: I think there may be more renames necessary for clarity so I decided not to bother with it in this PR and leave it for another one.

File: src/olm/account/one_time_keys.rs (hunk 2 above)
Comment (dkasak) [reply src/olm/account/one_time_keys.rs:35]: Yep, thanks for catching. What I actually should've done is remove the `Zeroize` impl for `OneTimeKeysPickle` to make everything symmetric. I'll make the change.

File: src/megolm/ratchet.rs (hunk 3 above)
Comment (dkasak) [reply src/megolm/ratchet.rs:219]: Added a TODO.

**Review by poljar — APPROVED** [review]

**Pull request merged**